source/blog/2016-06-14-rspec-3-5-has-been-released.md

```diff
@@ -0,0 +1,367 @@
+  automatically applied to including example groups. For example, maybe you want to temporarily
+  add `:skip` or `:focus` metadata to all including groups. There was no way to do this.
+* There's no obvious way to make a shared example group get auto-included in every
+  example group (e.g. for a global `before` hook or a `let` you want to make available everywhere...).
+* It's inconsistent with how module inclusion works (e.g. `config.include DBSupport, :db`).
+
+In RSpec 3.5 we've rectified these problems with a couple of changes.
+
+#### New API: `config.include_context`
+
+You can now define shared context inclusions in your `RSpec.configure` block:
+
+~~~ ruby
+RSpec.configure do |config|
+  config.include_context "DB support", :db
+end
+~~~
+
+This aligns with the existing `config.include` API for module inclusions, provides
+a way to include shared contexts based on metadata that is less surprising, and
+makes it simple to include a shared context in all example groups (just don't
+pass a metadata argument).
+
+#### New config option: `config.shared_context_metadata_behavior`
+
+We've also added a config option that lets you determine how shared context
+metadata is treated:
+
+~~~ ruby
+RSpec.configure do |config|
+  config.shared_context_metadata_behavior = :trigger_inclusion
+  # or
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+end
+~~~
+
+The former value (`:trigger_inclusion`) is the default and exists only for backwards
+compatibility. It treats metadata passed to `RSpec.shared_context` exactly how it was
+treated in RSpec 3.4 and before: it triggers inclusion in groups with matching metadata.
+We plan to remove support for it in RSpec 4.
+
+The latter value (`:apply_to_host_groups`) opts-in to the new behavior. Instead of
+triggering inclusion in groups with matching metadata, it applies the metadata to host
+groups.  For example, you could focus on all groups that use the DB by tagging your
+shared context:
+
+~~~ ruby
+RSpec.shared_context "DB support", :focus do
+  # ...
+end
+~~~
+
+### Expectations: Keyword argument support for the `respond_to` matcher.
+
+Keyword arguments have been a stable language feature for some time now but RSpec
+has lacked support for placing expectations upon keyword arguments in a variety of our matchers.
+
+In rspec-expectations 3.5 we add support for checking whether an object responds to a method
+using keyword arguments. You'll now be able to check
+
+~~~ ruby
+  expect(my_object).to respond_to(:find).with_keywords(:limit, :offset) }
+  expect(my_object).to respond_to(:find).with(1).argument.and_keywords(:limit, :offset)
+~~~
+
+ We're also expanding the matchers existing capabilities, adding the ability
+ to check for a range of arguments, or unlimited arguments in addition to a specific
+ number. e.g:
+
+~~~ ruby
+  expect(my_object).to respond_to(:build).with(2..3).arguments
+  expect(my_object).to respond_to(:build).with_unlimited_arguments
+~~~
```

Comment:
We should acknowledge the contributor who provided this feature.  In general, I try to always thank whoever built the feature.  The exception is myself...since I have usually written the blog posts I always thought it would be weird to thank myself so I leave that out.